i18n/README.md

```diff
@@ -5,11 +5,275 @@
+translator.t("apple_dish", mapOf("apples" to 3, "dishes" to 2)) // "With 3 apple(s), you can make 2 dish(es)."
+translator.t("indexed_trio", listOf("John", "Jane", "Jack")) // "Hello, John, Jane, and Jack!"
+translator.t("unnamed_trio", listOf("John", "Jane", "Jack")) // "Hello, John, Jane, and Jack!"
+```
+
+In fact, the rules for simple variable interpolation are the exact same as proposed by the 
+[Sprinkler Utils interpolation module](../utils/README.md#interpolation). Please refer to its documentation for more
+details.
+
+### Options
+
+TODO
+
+## Configuration
+
+As said in the [builder section](#translatorbuilder), the `TranslatorBuilder` has a method to set the configuration of
+the `Translator`: `TranslatorBuilder::withConfiguration`. This method takes in a `TranslatorConfiguration` object, which
+can be built with the `TranslatorConfiguration::create` DSL builder.
+
+Here is an example of a configuration with all the settings set to their default values (not setting the configuration,
+or not setting a specific property will result in the default value being used):
+
+```kt
+val translator = Translator.builder(Locale.ENGLISH)
+    ...
+    .withConfiguration(TranslatorConfiguration.create {
+        interpolationDelimiter = VariableDelimiter('{', '}')
+        pluralMapper = Plural.Mapper.defaultMapper()
+        formats = VariableFormatter.builtins()
+        missingKeyPolicy = MissingKeyPolicy.THROW_EXCEPTION
+    })
+    .build()
+```
+
+### `interpolationDelimiter`
+
+This property is used to set the symbols that will be used to delimit variables in translations. It is a
+`VariableDelimiter` object, with a constructor that takes in two `String`s: the opening and closing symbols.
+
+By default, the opening and closing symbols are `{` and `}` respectively.
+
+### `pluralMapper`
+
+This property is used to set the `Plural.Mapper` that will be used to determine the plural form of a translation. It is
+an interface with two methods which both take in a `Locale` and `Int` (which corresponds to the `count` option during
+pluralization). Both of these methods must return a `Plural` enum value, which follow and represent the 
+[CLDR plural rules](https://www.unicode.org/cldr/charts/latest/supplemental/language_plural_rules.html). That is to say,
+the possible values are:
+- `Plural.ZERO`
+- `Plural.ONE`
+- `Plural.TWO`
+- `Plural.FEW`
+- `Plural.MANY`
+- `Plural.OTHER`
+
+The two methods are `Plural.Mapper::mapPlural` and `Plural.Mapper::mapOrdinal`, which are used to determine the plural
+form of a translation, and the ordinal form of a translation respectively. The difference between the two is that the
+ordinal form is used when the translation is used to represent an ordinal number (for example, "1st", "2nd", "3rd", etc.)
+instead of a cardinal number (for example, "1", "2", "3", etc.).
+
+By default, the `Plural.Mapper.defaultMapper` method is used to create the `Plural.Mapper` object, which uses the
```

Comment:
No need to explain the implementation, you can link to the cldr rules for the english locale at most